In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Rectangle


## DATASET 1: CALIDAD DEL AGUA

In [2]:
pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')



aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

In [3]:
df = pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')


In [4]:
df.head(20)


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  copper  \
0        1.65    9.08    0.040    2.85    0.007        0.35      0.83    0.17   
1        2.32   21.16    0.010    3.31    0.002        5.28      0.68    0.66   
2        1.01   14.02    0.040    0.58    0.008        4.24      0.53    0.02   
3        1.36   11.33    0.040    2.96    0.001        7.23      0.03    1.66   
4        0.92   24.33    0.030    0.20    0.006        2.67      0.69    0.57   
5        0.94   14.47    0.030    2.88    0.003        0.80      0.43    1.38   
6        2.36     5.6    0.010    1.35    0.004        1.28      0.62    1.88   
7        3.93   19.87    0.040    0.66    0.001        6.22      0.10    1.86   
8        0.60   24.58    0.010    0.71    0.005        3.14      0.77    1.45   
9        0.22   16.76    0.020    1.37    0.007        6.40      0.49    0.82   
10       3.27     3.6    0.001    2.69    0.005        5.75      0.15    0.60   
11       1.35   21.96    0.040    0.84    0.002        0.10      0.76    0.17   
12       1.88   19.26    0.020    2.78    0.008        0.05      0.42    1.00   
13       4.93   23.98    0.040    3.05    0.008        0.70      0.51    1.35   
14       2.89   18.82    0.050    3.77    0.008        5.99      0.54    0.79   
15       0.61    2.41    0.030    0.59    0.002        1.94      0.77    1.54   
16       3.47   15.84    0.020    0.06    0.001        5.29      0.47    1.08   
17       2.11   17.03    0.020    0.88    0.009        7.78      0.88    1.15   
18       4.88   26.94    0.020    0.36    0.001        1.21      0.68    0.71   
19       4.12   17.99    0.020    3.43    0.006        0.01      0.41    1.82   

    flouride  bacteria  ...   lead  nitrates  nitrites  mercury  perchlorate  \
0       0.05      0.20  ...  0.054     16.08      1.13    0.007        37.75   
1       0.90      0.65  ...  0.100      2.01      1.93    0.003        32.26   
2       0.99      0.05  ...  0.078     14.16      1.11    0.006        50.28   
3       1.08      0.71  ...  0.016      1.41      1.29    0.004         9.12   
4       0.61      0.13  ...  0.117      6.74      1.11    0.003        16.90   
5       0.11      0.67  ...  0.135      9.75      1.89    0.006        27.17   
6       0.33      0.13  ...  0.021     18.60      1.78    0.007        45.34   
7       0.86      0.16  ...  0.197     13.65      1.81    0.001        53.35   
8       0.98      0.35  ...  0.167     14.66      1.84    0.004        23.43   
9       1.24      0.83  ...  0.109      4.79      1.46    0.010        30.42   
10      1.29      0.04  ...  0.145      8.47      1.25    0.006        55.40   
11      0.58      0.52  ...  0.011     18.40      1.49    0.009        21.52   
12      0.09      0.91  ...  0.103      4.37      1.95    0.006        22.12   
13      1.07      0.70  ...  0.101      1.16      1.11    0.008        26.80   
14      0.54      0.20  ...  0.126     17.56      1.82    0.009        17.54   
15      0.62      0.23  ...  0.017      1.99      1.08    0.007        11.16   
16      1.43      0.89  ...  0.080      1.91      1.20    0.008         0.18   
17      0.34      0.85  ...  0.065     17.86      1.53    0.003        19.40   
18      0.99      0.75  ...  0.071      0.31      1.22    0.002        56.70   
19      0.22      0.99  ...  0.108      8.06      1.76    0.005        24.29   

    radium  selenium  silver  uranium  is_safe  
0     6.78      0.08    0.34     0.02        1  
1     3.21      0.08    0.27     0.05        1  
2     7.07      0.07    0.44     0.01        0  
3     1.72      0.02    0.45     0.05        1  
4     2.41      0.02    0.06     0.02        1  
5     5.42      0.08    0.19     0.02        1  
6     2.84      0.10    0.24     0.08        0  
7     7.24      0.08    0.08     0.07        0  
8     4.99      0.08    0.25     0.08        1  
9     0.08      0.03    0.31     0.01        1  
10    7.80      0.05    0.33     0.06        0  
11    1.30      0.08    0.48     

In [5]:
df.tail(20)


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
7979       0.03   11.73     0.01    0.50     0.10        0.30      0.09   
7980       0.02    0.11     0.06    0.96     0.05        0.34      0.04   
7981       0.09    0.15     0.09    0.14     0.05        0.12      0.07   
7982       0.02    4.34     0.09    0.67     0.07        0.03      0.04   
7983       0.10    0.83     0.08    2.85     0.07        0.46      0.06   
7984       0.00    10.9     0.07    0.28     0.09        0.29      0.06   
7985       0.01    0.21     0.04    0.54     0.10        0.10      0.01   
7986       0.03   20.17     0.08    0.41     0.09        0.08      0.03   
7987       0.04    3.04     0.08    0.57     0.09        0.04      0.07   
7988       0.07    0.17     0.08    0.57     0.01        0.01      0.05   
7989       0.01   27.73     0.06    0.77     0.09        0.20      0.03   
7990       0.00    3.11     0.07    1.24     0.07        0.06      0.03   
7991       0.03    8.94     0.02    1.25     0.08        0.22      0.02   
7992       0.04   10.84     0.06    0.49     0.04        0.08      0.03   
7993       0.02    5.24     0.05    0.77     0.02        0.57      0.04   
7994       0.05    7.78     0.00    1.95     0.04        0.10      0.03   
7995       0.05   24.22     0.02    0.59     0.01        0.45      0.02   
7996       0.09    6.85     0.00    0.61     0.03        0.05      0.05   
7997       0.01      10     0.01    2.00     0.00        2.00      0.00   
7998       0.04    6.85     0.01    0.70     0.03        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
7979    0.06      1.43      0.16  ...  0.075     14.67       1.0    0.002   
7980    0.04      0.21      0.00  ...  0.111      4.74       1.0    0.001   
7981    0.07      1.06      0.00  ...  0.061      1.71       1.0    0.000   
7982    0.08      0.53      0.00  ...  0.159     15.32       1.0    0.006   
7983    0.01      0.11      0.00  ...  0.058      8.99       1.0    0.001   
7984    0.07      0.81      0.00  ...  0.066      5.51       1.0    0.000   
7985    0.01      1.30      0.00  ...  0.154     15.79       1.0    0.007   
7986    0.03      1.16      0.00  ...  0.114      0.83       1.0    0.005   
7987    0.06      0.96      0.00  ...  0.138      6.15       1.0    0.000   
7988    0.03      0.44      0.00  ...  0.047     14.61       1.0    0.002   
7989    0.00      0.58      0.00  ...  0.112     11.74       1.0    0.007   
7990    0.04      0.74      0.00  ...  0.016     15.15       1.0    0.005   
7991    0.05      0.71      0.00  ...  0.112     11.97       1.0    0.001   
7992    0.08      0.49      0.00  ...  0.151     14.44       1.0    0.003   
7993    0.03      0.15      0.00  ...  0.085     19.41       1.0    0.001   
7994    0.03      1.37      0.00  ...  0.197     14.29       1.0    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27       1.0    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92       1.0    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00       0.0    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92       1.0    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
7979         8.44    3.34      0.00    0.09     0.02        0  
7980        10.60    7.92      0.09    0.02     0.04        0  
7981         0.24    5.22      0.01    0.03     0.06        0  
7982         4.64    0.84      0.02    0.03     0.03        1  
7983         0.82    7.51      0.00    0.05     0.09        0  
7984         8.74    5.73      0.07    0.02     0.02        0  
7985         2.69    5.69      0.09    0.05     0.04        0  
7986         3.15    4.16      0.05    0.00     0.06        1  
7987         0.99    0.20      0.01    0.08     0.06        1  
7988         1.50    3.95      0.05    0.09     0.00        1  
7989         7.58    7.21      0.07    0.04     0.09        0  
7990         0.46    6.24      0.03    0.09 

In [23]:
df.sample(20)


aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
6267       0.07   10.12    0.060    1.13    0.030        0.37      0.08   
1803       4.41    29.4    0.490    4.25    0.010        1.65      0.88   
2722       0.03   22.79    0.020    0.45    0.005        7.08      0.62   
1313       0.54   28.08    0.270    1.01    0.110        2.22      0.65   
293        3.89    1.05    0.010    2.28    0.008        1.26      0.04   
5673       0.06    6.79    0.050    0.56    0.060        0.04      0.04   
7506       0.09    4.21    0.060    0.50    0.090        0.07      0.06   
3841       0.05   29.74    0.470    0.79    0.050        5.30      0.08   
5121       0.09   29.41    0.090    1.83    0.070        0.19      0.08   
4937       0.10    9.85    0.030    0.22    0.070        0.16      0.08   
3023       0.09   20.54    0.960    2.88    0.020        7.65      0.61   
3964       0.06   23.44    0.720    3.19    0.130        7.12      0.40   
7728       0.00    2.88    0.020    2.31    0.030        0.29      0.03   
2542       0.01   26.55    0.001    3.08    0.007        7.87      0.37   
3148       0.01    9.24    0.290    2.14    0.060        7.61      0.19   
5891       0.10      17    0.060    1.23    0.070        0.03      0.04   
1911       3.76    6.42    0.570    0.48    0.050        0.40      0.47   
4286       0.03   23.91    0.030    2.55    0.030        0.07      0.01   
1297       0.24       5    0.310    2.56    0.100        7.89      0.05   
3255       0.07    7.79    0.870    2.76    0.080        6.74      0.06   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
6267    1.07      1.35      0.39  ...  0.113     14.25      1.24    0.003   
1803    1.19      0.38      0.84  ...  0.064      8.12      1.16    0.005   
2722    0.61      1.46      0.84  ...  0.136     16.56      1.62    0.007   
1313    0.62      0.44      0.00  ...  0.136     12.40      1.34    0.004   
293     0.14      0.77      0.24  ...  0.062     13.59      1.98    0.005   
5673    1.47      0.37      0.74  ...  0.130     17.77      0.46    0.006   
7506    1.99      0.81      0.00  ...  0.098      4.75      1.66    0.003   
3841    0.09      0.69      0.77  ...  0.110      0.85      1.33    0.003   
5121    1.16      1.06      0.00  ...  0.120      7.09      0.26    0.001   
4937    0.01      0.90      0.00  ...  0.014     17.66      0.99    0.007   
3023    1.23      1.44      0.56  ...  0.012     10.12      2.07    0.004   
3964    0.03      0.52      0.49  ...  0.057     13.89      1.51    0.005   
7728    0.09      0.91      0.00  ...  0.191      1.81      1.42    0.005   
2542    1.75      0.19      0.90  ...  0.045      4.35      1.32    0.007   
3148    1.22      1.06      0.27  ...  0.103      3.71      1.89    0.007   
5891    0.87      0.30      0.76  ...  0.022      7.12      0.07    0.009   
1911    1.58      0.43      0.98  ...  0.187      9.61      2.68    0.001   
4286    0.04      0.07      0.00  ...  0.042     16.71      0.24    0.009   
1297    0.15      1.19      0.00  ...  0.164     17.72      1.56    0.001   
3255    0.67      0.63      0.48  ...  0.007      9.53      1.32    0.002   

      perchlorate  radium  selenium  silver  uranium  is_safe  
6267         0.38    4.33      0.06    0.07     0.04        0  
1803         7.10    3.72      0.07    0.42     0.06        0  
2722        37.96    5.67      0.06    0.15     0.08        0  
1313        33.81    2.12      0.09    0.37     0.02        0  
293         51.76    0.99      0.08    0.49     0.08        1  
5673         0.23    0.97      0.06    0.00     0.03        0  
7506         4.38    0.06      0.10    0.06     0.02        0  
3841        33.74    3.34      0.06    0.48     0.04        0  
5121         2.17    3.50      0.04    0.04     0.04        0  
4937         6.04    2.78      0.10    0.04     0.04        0  
3023        56.40    3.22      0.07    0.18     0.08        0  
3964        30.57    3.10      0.07    0.31 

#### VALORES FALTANTES DEL DATASET

In [6]:
df.dropna()

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

In [25]:
df.fillna(0)

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

### SOBRE EL DATASET

CONTEXTO ANALITICO

    El Dataset con el que se trabajará a continuación corresponde a un conjunto de datos referidos al contenido de minerales del agua en un pueblo llamado Dioncaster, el cual se ubica en el extremo occidental del condado de Sabbatown.
    El archivo .csv utilizado fue descargado de la página kaggle.com.

CONTEXTO COMERCIAL

    El objetivo del presente trabajo es analizar como impacta el contenido de determinados minerales en la calidad de agua, la cual es utilizada tanto para riego de cultivos, como para consumo animal. Por lo cual se pretende discriminar en cuanto a la aptitud del agua, en base al fin con el que se la vaya a utilizar.

DESCRIPCIÓN  

Todos los atributos del dataset listados a continuación están expresados por litro:
- aluminium: aluminio
- arsenic: arsénico
- barium: bario
- cadmium: cadmio
- chloramine: cloramina
- chromium: cromo
- cooper: cobre
- fluoride: fluoruro
- bacteria: bacterias
- viruses: virus
- lead: plomo
- nitrates: nitratos
- nitrites: nitritos
- mercury: mercurio
- perchlorate: perclorato
- radium: radio
- selenium: selenio
- silver: plata
- uranium: uranio
- is_safe: es seguro para consumo



El dataset seleccionado cuenta con 7999 filas y 21 columnas

In [7]:
display(df.shape)


(7999, 21)

El dataset cuenta con los siguientes nombres en sus columnas

In [8]:
df.columns

Index(['aluminium', 'ammonia', 'arsenic', 'barium', 'cadmium', 'chloramine',
       'chromium', 'copper', 'flouride', 'bacteria', 'viruses', 'lead',
       'nitrates', 'nitrites', 'mercury', 'perchlorate', 'radium', 'selenium',
       'silver', 'uranium', 'is_safe'],
      dtype='object')

Bad pipe message: %s [b'\x06\xaeo4\xc8\x99\xda\xb6\x15\xe2\xeaU\xa0\x9f\x0chS\xa7 x\x0e\x0b\xf4@\xaf\x89X\xf0\x14k\x10\xb8\x9cE\xec/\xa3\xcf\x8a\xa3]1\xae\xfc\xdb\x9e\x11\xe4\xa2\x89c\x00\x08\x13\x02\x13\x03\x13\x01\x00\xff\x01\x00\x00\x8f\x00\x00\x00\x0e\x00\x0c\x00\x00\t127.0.0.1\x00\x0b\x00\x04\x03\x00\x01']
Bad pipe message: %s [b'\n']
Bad pipe message: %s [b'\xa5\xdf\x8fI\xef\xc8s\xe04\xb6\x96\xb6\x9b\x90\xa6\xb1\xc6K n]', b'i2A\xcc\x86w\xcfbk}\xc8\xcc=V\xa5+\x1a\x110\x9e\x83\xc0\xcb\xd6\xfd4\xa6\x84\x00\x08\x13\x02\x13\x03\x13\x01\x00\xff\x01\x00\x00\x8f\x00\x00\x00\x0e\x00\x0c\x00\x00\t127.0.0.1\x00\x0b\x00\x04\x03\x00\x01\x02\x00\n\x00\x0c\x00\n\x00\x1d\x00\x17\x00\x1e\x00\x19\x00\x18\x00#\x00\x00\x00\x16\x00\x00\x00\x17\x00\x00\x00\r\x00\x1e\x00\x1c\x04\x03\x05\x03\x06\x03\x08\x07\x08\x08\x08\t\x08\n\x08\x0b\x08\x04\x08\x05\x08\x06\x04\x01\x05\x01\x06\x01\x00+\x00\x03\x02\x03\x04\x00-\x00\x02\x01\x01\x003\x00&\x00$\x00']
Bad pipe message: %s [b' 3o\xf4G(e\xf3?\x04q\x0c\xe2\xdf

El dataset cuenta con los siguientes tipos de datos en sus columnas

In [13]:
df.dtypes

aluminium      float64
ammonia         object
arsenic        float64
barium         float64
cadmium        float64
chloramine     float64
chromium       float64
copper         float64
flouride       float64
bacteria       float64
viruses        float64
lead           float64
nitrates       float64
nitrites       float64
mercury        float64
perchlorate    float64
radium         float64
selenium       float64
silver         float64
uranium        float64
is_safe         object
dtype: object

La información proporcionada es la siguiente:

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7999 entries, 0 to 7998
Data columns (total 21 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   aluminium    7999 non-null   float64
 1   ammonia      7999 non-null   object 
 2   arsenic      7999 non-null   float64
 3   barium       7999 non-null   float64
 4   cadmium      7999 non-null   float64
 5   chloramine   7999 non-null   float64
 6   chromium     7999 non-null   float64
 7   copper       7999 non-null   float64
 8   flouride     7999 non-null   float64
 9   bacteria     7999 non-null   float64
 10  viruses      7999 non-null   float64
 11  lead         7999 non-null   float64
 12  nitrates     7999 non-null   float64
 13  nitrites     7999 non-null   float64
 14  mercury      7999 non-null   float64
 15  perchlorate  7999 non-null   float64
 16  radium       7999 non-null   float64
 17  selenium     7999 non-null   float64
 18  silver       7999 non-null   float64
 19  uraniu

### La variable que se busca predecir es la llamada "is_safe" a=is_safe 
### Es decir si el agua es apta para consumo animal.


#### Las variables consideradas relevantes para predecir la aptitud en consumo animal son z1=arsenic z2=fluorides z3=nitrates

#### z1=arsenic y z2=fluorides se ha seleccionado ya que el As y el F por si mismos pueden definir la ineptitud del agua para ser consumida. SOn elementos altamente tóxicos para animales y para el hombre, por lo que independientemente de la composición salina del agua en bebida, altos niveles de uno u otro limitan su uso.
#### z3=nitrates se ha elegido ya que los nitratos se consideran peligrosos a partir de 200 ppm y son indicadores de contaminación orgánica, también se ha encontrado que están muy relacionados a la fertilización irracional con urea.

In [27]:
print (type(df['is_safe']),type(df[['is_safe']]))
display(df['is_safe'].head(20))
display(df[['is_safe']].head(20))


<class 'pandas.core.series.Series'> <class 'pandas.core.frame.DataFrame'>


0     1
1     1
2     0
3     1
4     1
5     1
6     0
7     0
8     1
9     1
10    0
11    1
12    1
13    1
14    1
15    1
16    1
17    1
18    0
19    1
Name: is_safe, dtype: object

is_safe
0        1
1        1
2        0
3        1
4        1
5        1
6        0
7        0
8        1
9        1
10       0
11       1
12       1
13       1
14       1
15       1
16       1
17       1
18       0
19       1

## DATASET 2: PREDICCIÓN TITULOS SIMILARES

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Rectangle

In [6]:
pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/netflix_titles.csv')


show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

In [7]:
df = pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/netflix_titles.csv')



In [8]:
df.head(10)


show_id     type                             title  \
0      s1    Movie              Dick Johnson Is Dead   
1      s2  TV Show                     Blood & Water   
2      s3  TV Show                         Ganglands   
3      s4  TV Show             Jailbirds New Orleans   
4      s5  TV Show                      Kota Factory   
5      s6  TV Show                     Midnight Mass   
6      s7    Movie  My Little Pony: A New Generation   
7      s8    Movie                           Sankofa   
8      s9  TV Show     The Great British Baking Show   
9     s10    Movie                      The Starling   

                        director  \
0                Kirsten Johnson   
1                            NaN   
2                Julien Leclercq   
3                            NaN   
4                            NaN   
5                  Mike Flanagan   
6  Robert Cullen, José Luis Ucha   
7                   Haile Gerima   
8                Andy Devonshire   
9                 Theodore Melfi   

                                                cast  \
0                                                NaN   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
3                                                NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...   
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   

                                             country          date_added  \
0                                      United States  September 25, 2021   
1                                       South Africa  September 24, 2021   
2                                                NaN  September 24, 2021   
3                                                NaN  September 24, 2021   
4                                              India  September 24, 2021   
5                                                NaN  September 24, 2021   
6                                                NaN  September 24, 2021   
7  United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                     United Kingdom  September 24, 2021   
9                                      United States  September 24, 2021   

   release_year rating   duration  \
0          2020  PG-13     90 min   
1          2021  TV-MA  2 Seasons   
2          2021  TV-MA   1 Season   
3          2021  TV-MA   1 Season   
4          2021  TV-MA  2 Seasons   
5          2021  TV-MA   1 Season   
6          2021     PG     91 min   
7          1993  TV-MA    125 min   
8          2021  TV-14  9 Seasons   
9          2021  PG-13    104 min   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   
5                 TV Dramas, TV Horror, TV Mysteries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
8                       British TV Shows, Reality TV   
9                                   Comedies, Dramas   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...  
5  The arrival of a charismatic young priest brin...  
6  Equestria's divided. But a bright-eyed hero be...  
7  On a photo shoot in Ghana, an American model s...  
8  A talented 

In [32]:
df.tail(10)


show_id     type               title                 director  \
8797   s8798  TV Show           Zak Storm                      NaN   
8798   s8799    Movie            Zed Plus  Chandra Prakash Dwivedi   
8799   s8800    Movie               Zenda           Avadhoot Gupte   
8800   s8801  TV Show  Zindagi Gulzar Hai                      NaN   
8801   s8802    Movie             Zinzana          Majid Al Ansari   
8802   s8803    Movie              Zodiac            David Fincher   
8803   s8804  TV Show         Zombie Dumb                      NaN   
8804   s8805    Movie          Zombieland          Ruben Fleischer   
8805   s8806    Movie                Zoom             Peter Hewitt   
8806   s8807    Movie              Zubaan              Mozez Singh   

                                                   cast  \
8797  Michael Johnston, Jessica Gee-George, Christin...   
8798  Adil Hussain, Mona Singh, K.K. Raina, Sanjay M...   
8799  Santosh Juvekar, Siddharth Chandekar, Sachit P...   
8800  Sanam Saeed, Fawad Khan, Ayesha Omer, Mehreen ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8803                                                NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                            country          date_added  \
8797  United States, France, South Korea, Indonesia  September 13, 2018   
8798                                          India   December 31, 2019   
8799                                          India   February 15, 2018   
8800                                       Pakistan   December 15, 2016   
8801                   United Arab Emirates, Jordan       March 9, 2016   
8802                                  United States   November 20, 2019   
8803                                            NaN        July 1, 2019   
8804                                  United States    November 1, 2019   
8805                                  United States    January 11, 2020   
8806                                          India       March 2, 2019   

      release_year rating   duration  \
8797          2016  TV-Y7  3 Seasons   
8798          2014  TV-MA    131 min   
8799          2009  TV-14    120 min   
8800          2012  TV-PG   1 Season   
8801          2015  TV-MA     96 min   
8802          2007      R    158 min   
8803          2018  TV-Y7  2 Seasons   
8804          2009      R     88 min   
8805          2006     PG     88 min   
8806          2015  TV-14    111 min   

                                              listed_in  \
8797                                           Kids' TV   
8798             Comedies, Dramas, International Movies   
8799                       Dramas, International Movies   
8800  International TV Shows, Romantic TV Shows, TV ...   
8801            Dramas, International Movies, Thrillers   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
8797  Teen surfer Zak Storm is mysteriously transpor...  
8798  A philandering small-town mechanic's political...  
8799  A change in the leadership of a political part...  
8800  Strong-willed, middle-class Kashaf and carefre...  
8801  Recovering alcoholic Talal wakes up inside a s...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...

In [34]:
df.sample(10)


show_id   type                   title                     director  \
3768   s3769  Movie                   Stree                 Amar Kaushik   
3154   s3155  Movie  Seven and a half dates               Biodun Stephen   
7726   s7727  Movie     People You May Know                  J.C. Falcón   
6203   s6204  Movie               B.A. Pass                    Ajay Bahl   
489     s490  Movie             Back to Q82               Ahmed Siddiqui   
7269   s7270  Movie              Last Ferry                 Jaki Bradley   
6175   s6176  Movie                Antidote                   Ken Barbet   
7427   s7428  Movie                 Maynard            Samuel D. Pollard   
5131   s5132  Movie                 Shikari  Latif Faiziyev, Umesh Mehra   
4993   s4994  Movie            The Outsider             Martin Zandvliet   

                                                   cast  \
3768  Rajkummar Rao, Shraddha Kapoor, Pankaj Tripath...   
3154  Mercy Johnson, Jim Iyke, Toyin Abraham, Sola S...   
7726  Sean Maher, Andrea Grano, Mark Cirillo, Nacho ...   
6203  Shadab Kamal, Shilpa Shukla, Rajesh Sharma, Di...   
489   Jassim Al Nabhan, Abdel Imam Abdullah, Abdulaz...   
7269  Ramón O. Torres, Sheldon Best, Myles Clohessy,...   
6175  Randy Couture, Chuck Zito, Wil Traval, Celeste...   
7427                                                NaN   
5131  Mithun Chakraborty, Naseeruddin Shah, Irina Ku...   
4993  Jared Leto, Tadanobu Asano, Kippei Shiina, Shi...   

                    country         date_added  release_year rating duration  \
3768                  India       June 7, 2019          2018  TV-MA  127 min   
3154                    NaN  December 13, 2019          2018  TV-14  107 min   
7726   United States, Spain  February 28, 2017          2016  TV-MA  108 min   
6203                  India       June 7, 2018          2013  TV-14   99 min   
489   Kuwait, United States       July 8, 2021          2017  TV-14  104 min   
7269          United States     March 11, 2020          2019  TV-MA   86 min   
6175          United States     April 14, 2019          2018  TV-MA   90 min   
7427          United States  September 2, 2018          2017  TV-PG  100 min   
5131    India, Soviet Union   December 8, 2017          1991  TV-14  163 min   
4993          United States      March 9, 2018          2018  TV-MA  121 min   

                                             listed_in  \
3768       Comedies, Horror Movies, Independent Movies   
3154     Dramas, International Movies, Romantic Movies   
7726                              Dramas, LGBTQ Movies   
6203  Dramas, Independent Movies, International Movies   
489                     Comedies, International Movies   
7269          Dramas, Independent Movies, LGBTQ Movies   
6175                 Action & Adventure, Horror Movies   
7427                                     Documentaries   
5131    Dramas, International Movies, Music & Musicals   
4993                                 Dramas, Thrillers   

                                            description  
3768  Once a year, men in a small town fear abductio...  
3154  Engrossed in her career, a young woman relucta...  
7726  The lives of several friends are thrown into c...  
6203  Orphaned and at loose ends, young Mukesh falls...  
489   Eager to rejoin her divorced parents, an inven...  
7269  Seeking romance and friendship, a young gay la...  
6175  A tough-as-nails treasure hunter protects a hu...  
7427  Friends, family and political titans recall th...  
5131  When Shankar, who performs on the road with pa...  
4993  An American soldier imprisoned in postwar Japa...

#### VALORES FALTANTES DEL DATASET

In [35]:
df.dropna()

show_id     type                          title             director  \
7         s8    Movie                        Sankofa         Haile Gerima   
8         s9  TV Show  The Great British Baking Show      Andy Devonshire   
9        s10    Movie                   The Starling       Theodore Melfi   
12       s13    Movie                   Je Suis Karl  Christian Schwochow   
24       s25    Movie                          Jeans           S. Shankar   
...      ...      ...                            ...                  ...   
8801   s8802    Movie                        Zinzana      Majid Al Ansari   
8802   s8803    Movie                         Zodiac        David Fincher   
8804   s8805    Movie                     Zombieland      Ruben Fleischer   
8805   s8806    Movie                           Zoom         Peter Hewitt   
8806   s8807    Movie                         Zubaan          Mozez Singh   

                                                   cast  \
7     Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8     Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12    Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   
24    Prashanth, Aishwarya Rai Bachchan, Sri Lakshmi...   
...                                                 ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                                country          date_added  \
7     United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                        United Kingdom  September 24, 2021   
9                                         United States  September 24, 2021   
12                              Germany, Czech Republic  September 23, 2021   
24                                                India  September 21, 2021   
...                                                 ...                 ...   
8801                       United Arab Emirates, Jordan       March 9, 2016   
8802                                      United States   November 20, 2019   
8804                                      United States    November 1, 2019   
8805                                      United States    January 11, 2020   
8806                                              India       March 2, 2019   

      release_year rating   duration  \
7             1993  TV-MA    125 min   
8             2021  TV-14  9 Seasons   
9             2021  PG-13    104 min   
12            2021  TV-MA    127 min   
24            1998  TV-14    166 min   
...            ...    ...        ...   
8801          2015  TV-MA     96 min   
8802          2007      R    158 min   
8804          2009      R     88 min   
8805          2006     PG     88 min   
8806          2015  TV-14    111 min   

                                             listed_in  \
7     Dramas, Independent Movies, International Movies   
8                         British TV Shows, Reality TV   
9                                     Comedies, Dramas   
12                        Dramas, International Movies   
24     Comedies, International Movies, Romantic Movies   
...                                                ...   
8801           Dramas, International Movies, Thrillers   
8802                    Cult Movies, Dramas, Thrillers   
8804                           Comedies, Horror Movies   
8805                Children & Family Movies, Comedies   
8806    Dramas, International Movies, Music & Musicals   

                                            description  
7     On a photo shoot in Ghana, an American model s...  
8     A talented batch of amateur bakers face off in...  
9     A woman adjusting to life after a loss contend...  
12    After 

In [36]:
df.fillna(0)

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water                0   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans                0   
4         s5  TV Show           Kota Factory                0   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb                0   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                     0  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...              0   
3                                                     0              0   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                  0              0   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

### SOBRE EL DATASET

CONTEXTO ANALITICO

    Netflix es una de las plataformas de transmisión de video y medios mas populares. Posee mas de 8000 películas y programas de TV disponibles en su plataforma, y mas de 200 millones de suscriptores en todo el mundo. Este dataset reune información sobre peliculas y programas de tv disponibles en Netflix.

    El archivo .csv utilizado fue descargado de la página kaggle.com.

CONTEXTO COMERCIAL

    El objetivo del presente trabajo es optimizar el algoritmo de la plataforma, para que las recomendaciones propuestas se ajusten a los deseos de los usuarios.

DESCRIPCIÓN  

Todos los atributos del dataset listados se presentan a continuación:
- show_id: ID unico para cada pelicula/programa de tv
- type: identificador de pelicula/programa de tv
- title: titulo de pelicula/programa de tv
- Director: director de pelicula/programa de tv
- cast: actores involucrados en la pelicula/programa de tv
- date_added: fecha en la cual se agregó a Netflix
- release_year: año de lanzamiento de pelicula/programa de tv
- rating: calificacion de pelicula/programa de tv según edad
- duration: duración total en minutos o numero de temporadas de pelicula/programa de tv
- listed_in: clasificación en base a categoría de pelicula/programa de tv
-description: sinopsis de pelicula/programa de tv



El dataset seleccionado cuenta con 8807 filas y 12 columnas

In [9]:
display(df.shape)

(8807, 12)

El dataset cuenta con los siguientes nombres en sus columnas:

In [38]:
df.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')

El dataset cuenta con los siguientes tipos de datos en sus columnas:

In [39]:
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

La información proporcionada es la siguiente:

In [40]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


### La variable que se busca predecir es "Similar Content" x=similar_content 
### Es decir encontrar la manera de que la variable se ajuste a los requerimientos de usuario, basada en características relevantes.


#### Las variables consideradas relevantes para predecir el contenido similar son a1=listed_in a2=rating a3=director

#### a1=listed_in se ha seleccionado ya que se identifica como de suma importancia para precedir un contenido similar, ajustarse a si lo que desea ver el suscriptor va a ser clasificado de la misma manera que lo que lo llevó a ver determinada pelicula o programa de tv 
#### a2=rating se utiliza ya que la categoría de película/programa de tv en base a quién esta apto para ver dicho contenido hace pensar que si el usuario busca contenido similar, no debería salir de dicha categoria. Por otro lado, es importante en caso de menores de edad, que no se recomiende algo no apto para dichi público.
#### a3=director el director suele tener su propia impronta y manera de dirigir el contenido, por lo cual se cree relevante que si a un usuario le gustó determinado contenido, seleccionar el mismo director o alguno similar puede satisfacer los requerimientos.

## DATASET 3: PREDICIÓN DE GANADORES EN TENIS DE MESA

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Rectangle

In [11]:
pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/tenisdemesa.csv')

Unnamed: 0                 Date         Player1         Player2  \
0              1  2022-06-10 01:05:00      Marchuk M.  Polysynskyi M.   
1              2  2022-06-10 01:25:00      Sokolov S.       Sachuk S.   
2              3  2022-06-10 01:35:00      Samofal Y.     Bambutsa R.   
3              4  2022-06-10 01:55:00        Kyilo V.   Chernyshov D.   
4              5  2022-06-10 02:05:00  Polysynskyi M.      Kostiuk B.   
...          ...                  ...             ...             ...   
7846        7847  2022-07-08 13:50:00        Rehak M.       Bazany D.   
7847        7848  2022-07-08 13:55:00     Bielokon Y.       Yushko R.   
7848        7849  2022-07-08 13:55:00      Kuzmych V.          Rak S.   
7849        7850  2022-07-08 14:00:00   Bondarenko I.    Havryliuk B.   
7850        7851  2022-07-08 14:00:00    Kurylenko O.     Kuzmenko D.   

      Sets_P1  Sets_P2  P1_G1  P2_G1  P1_G2  P2_G2  P1_G3  P2_G3  P1_G4  \
0           3        1     11      8     11      6      6     11   11.0   
1           0        3     12     14      8     11      5     11    NaN   
2           3        1      4     11     11      5     11      6   11.0   
3           3        1      6     11     11      1     15     13   11.0   
4           3        0     12     10     11      9     11      6    NaN   
...       ...      ...    ...    ...    ...    ...    ...    ...    ...   
7846        3        2      4     11     11      9      4     11   11.0   
7847        3        2     11      6     11      8      8     11   10.0   
7848        2        3     11      9      3     11     13     11    8.0   
7849        0        3     10     12      8     11      4     11    NaN   
7850        2        3     11      7      5     11     12     10    6.0   

      P2_G4  P1_G5  P2_G5  HomeWinner  
0       8.0    NaN    NaN           1  
1       NaN    NaN    NaN           0  
2       9.0    NaN    NaN           1  
3       7.0    NaN    NaN           1  
4       NaN    NaN    NaN           1  
...     ...    ...    ...         ...  
7846    5.0   11.0    7.0           1  
7847   12.0   11.0    3.0           1  
7848   11.0    5.0   11.0           0  
7849    NaN    NaN    NaN           0  
7850   11.0    9.0   11.0           0  

[7851 rows x 17 columns]

In [12]:
df = pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/tenisdemesa.csv')

In [43]:
df.head(20)

Unnamed: 0                 Date         Player1         Player2  Sets_P1  \
0            1  2022-06-10 01:05:00      Marchuk M.  Polysynskyi M.        3   
1            2  2022-06-10 01:25:00      Sokolov S.       Sachuk S.        0   
2            3  2022-06-10 01:35:00      Samofal Y.     Bambutsa R.        3   
3            4  2022-06-10 01:55:00        Kyilo V.   Chernyshov D.        3   
4            5  2022-06-10 02:05:00  Polysynskyi M.      Kostiuk B.        3   
5            6  2022-06-10 02:30:00       Sachuk S.    Diachenko M.        2   
6            7  2022-06-10 02:35:00   Balakiriev M.      Marchuk M.        3   
7            8  2022-06-10 02:55:00  Hryshchenko O.      Sokolov S.        3   
8            9  2022-06-10 03:05:00      Samofal Y.  Polysynskyi M.        0   
9           10  2022-06-10 03:25:00        Kyilo V.       Sachuk S.        2   
10          11  2022-06-10 03:35:00      Marchuk M.     Bambutsa R.        3   
11          12  2022-06-10 03:55:00      Sokolov S.   Chernyshov D.        3   
12          13  2022-06-10 04:05:00   Balakiriev M.      Kostiuk B.        3   
13          14  2022-06-10 04:25:00  Hryshchenko O.    Diachenko M.        3   
14          15  2022-06-10 04:50:00      Samofal Y.      Marchuk M.        0   
15          16  2022-06-10 04:55:00        Kyilo V.      Sokolov S.        2   
16          17  2022-06-10 05:10:00   Balakiriev M.  Polysynskyi M.        3   
17          18  2022-06-10 05:25:00  Hryshchenko O.       Sachuk S.        3   
18          19  2022-06-10 05:35:00     Bambutsa R.      Kostiuk B.        3   
19          20  2022-06-10 05:55:00   Chernyshov D.    Diachenko M.        1   

    Sets_P2  P1_G1  P2_G1  P1_G2  P2_G2  P1_G3  P2_G3  P1_G4  P2_G4  P1_G5  \
0         1     11      8     11      6      6     11   11.0    8.0    NaN   
1         3     12     14      8     11      5     11    NaN    NaN    NaN   
2         1      4     11     11      5     11      6   11.0    9.0    NaN   
3         1      6     11     11      1     15     13   11.0    7.0    NaN   
4         0     12     10     11      9     11      6    NaN    NaN    NaN   
5         3     11      9     11      9      5     11   10.0   12.0    0.0   
6         1     11      8     11      7      3     11   11.0    9.0    NaN   
7         2     11      8      7     11     11      3    9.0   11.0   11.0   
8         3      9     11      9     11      8     11    NaN    NaN    NaN   
9         3      9     11      5     11     11      7   11.0    6.0    9.0   
10        0     11      6     13     11     11      3    NaN    NaN    NaN   
11        2     11      7     12     10      5     11    6.0   11.0   11.0   
12        2     11      5     13     11      7     11    8.0   11.0   11.0   
13        2     11      7     11      5      9     11    8.0   11.0   12.0   
14        3      3     11      3     11      9     11    NaN    NaN    NaN   
15        3     10     12      8     11     11      7   11.0    5.0    8.0   
16        2     11      5     11      8      9     11    6.0   11.0   11.0   
17        2     11      9     11      5     13     15    6.0   11.0   11.0   
18        1      7     11     13     11     11      8   11.0    8.0    NaN   
19        3     12     10      8     11      3     11    6.0   11.0    NaN   

    P2_G5  HomeWinner  
0     NaN           1  
1     NaN           0  
2     NaN           1  
3     NaN           1  
4     NaN           1  
5    11.0           0  
6     NaN           1  
7     9.0           1  
8     NaN           0  
9    11.0           0  
10    NaN           1  
11    9.0           1  
12    8.0           1  
13   10.0           1  
14    NaN           0  
15   11.0           0  
16    5.0           1  
17    9.0           1  
18    NaN           1  
19    NaN           0

In [44]:
df.tail(20)

Unnamed: 0                 Date            Player1        Player2  \
7831        7832  2022-07-08 13:10:00       Chornobab V.     Honchar Y.   
7832        7833  2022-07-08 13:15:00          Blazek J.       Barsa T.   
7833        7834  2022-07-08 13:20:00          Lenart R.    Cechvala R.   
7834        7835  2022-07-08 13:20:00         Lutskyi V.     Maltsev D.   
7835        7836  2022-07-08 13:20:00           Zotov Y.  Plamadeala D.   
7836        7837  2022-07-08 13:25:00         Nemtsov S.       Torba R.   
7837        7838  2022-07-08 13:25:00      Peretiatko A.      Husiev V.   
7838        7839  2022-07-08 13:30:00  Miroshnychenko S.       Tupyk M.   
7839        7840  2022-07-08 13:30:00      Seluchenko O.  Vasylyshyn I.   
7840        7841  2022-07-08 13:35:00     Bryzhevatyi O.   Kunchurov D.   
7841        7842  2022-07-08 13:40:00         Kovshun Y.     Solomko S.   
7842        7843  2022-07-08 13:40:00       Yurchenko K.    Zholubak R.   
7843        7844  2022-07-08 13:45:00           Lukes T.      Herink J.   
7844        7845  2022-07-08 13:50:00          Kostyv M.   Motrunych I.   
7845        7846  2022-07-08 13:50:00     Pisceanschi V.     Dubchak I.   
7846        7847  2022-07-08 13:50:00           Rehak M.      Bazany D.   
7847        7848  2022-07-08 13:55:00        Bielokon Y.      Yushko R.   
7848        7849  2022-07-08 13:55:00         Kuzmych V.         Rak S.   
7849        7850  2022-07-08 14:00:00      Bondarenko I.   Havryliuk B.   
7850        7851  2022-07-08 14:00:00       Kurylenko O.    Kuzmenko D.   

      Sets_P1  Sets_P2  P1_G1  P2_G1  P1_G2  P2_G2  P1_G3  P2_G3  P1_G4  \
7831        1        3      7     11     12     10      7     11    9.0   
7832        1        3     11      4      6     11      2     11    4.0   
7833        3        1      3     11     11      9     11      7   12.0   
7834        1        3      6     11     11      3      6     11    9.0   
7835        0        3      9     11      7     11      3     11    NaN   
7836        1        3      9     11     11      7      6     11    5.0   
7837        3        1     14     12     11      6      7     11   11.0   
7838        1        3     11      4      4     11      7     11    8.0   
7839        3        0     12     10     11      9     11      7    NaN   
7840        0        3      7     11      8     11     12     14    NaN   
7841        2        3      9     11     11      9      9     11   11.0   
7842        2        3     11      9      7     11     11      6    7.0   
7843        1        3      4     11     11      3     11     13    8.0   
7844        3        0     13     11     11      9     11      8    NaN   
7845        1        3      5     11      2     11     11      2    7.0   
7846        3        2      4     11     11      9      4     11   11.0   
7847        3        2     11      6     11      8      8     11   10.0   
7848        2        3     11      9      3     11     13     11    8.0   
7849        0        3     10     12      8     11      4     11    NaN   
7850        2        3     11      7      5     11     12     10    6.0   

      P2_G4  P1_G5  P2_G5  HomeWinner  
7831   11.0    NaN    NaN           0  
7832   11.0    NaN    NaN           0  
7833   10.0    NaN    NaN           1  
7834   11.0    NaN    NaN           0  
7835    NaN    NaN    NaN           0  
7836   11.0    NaN    NaN           0  
7837    7.0    NaN    NaN           1  
7838   11.0    NaN    NaN           0  
7839    NaN    NaN    NaN           1  
7840    NaN    NaN    NaN           0  
7841    6.0   10.0   12.0           0  
7842   11.0   12.0   14.0           0  
7843   11.0    NaN    NaN           0  
7844    NaN    NaN    NaN           1  
7845   11.0    NaN    NaN           0  
7846    5.0   11.0    7.0           1  
7847   12.0   11.0    3.0           1  
7848   11.0    5.0   11.0           0  
7849    NaN    NaN    NaN           0  
7850   11.0    9.0   11.0           0

In [45]:
df.sample(20)


Unnamed: 0                 Date         Player1         Player2  \
7168        7169  2022-07-06 09:55:00       Pyrtyk B.    Nikolenko A.   
1              2  2022-06-10 01:25:00      Sokolov S.       Sachuk S.   
7381        7382  2022-07-06 22:55:00    Chepurnyi A.   Hrymovskyi O.   
1483        1484  2022-06-15 10:20:00       Gogora M.        Bucko P.   
6130        6131  2022-07-02 12:40:00       Ziakun V.   Kolomiiets V.   
3116        3117  2022-06-21 09:20:00     Veternik P.     Papirnik O.   
4013        4014  2022-06-24 10:00:00    Yakimenko S.     Stukalov V.   
601          602  2022-06-12 09:20:00  Dobrzhinsky O.       Shkepu S.   
4460        4461  2022-06-25 20:55:00      Pavlyuk O.      Kravets Y.   
758          759  2022-06-12 18:25:00      Tkachov S.      Babenko P.   
3592        3593  2022-06-22 20:35:00       Melnik V.     Cherevko R.   
5879        5880  2022-07-01 11:55:00     Vlasenko V.        Torba R.   
7204        7205  2022-07-06 12:05:00     Stukalov V.    Kunchurov D.   
37            38  2022-06-10 08:25:00          Rak S.  Nakliutskyi A.   
4753        4754  2022-06-27 01:00:00    Nikolenko S.      Sydorak N.   
6841        6842  2022-07-05 07:25:00        Masko Y.     Cherevko R.   
5358        5359  2022-06-29 11:50:00     Martinka S.        Janik A.   
3827        3828  2022-06-23 16:20:00      Capusta A.        Bulat P.   
6083        6084  2022-07-02 09:20:00       Krupii O.       Dubyna I.   
2504        2505  2022-06-19 00:05:00      Gomilko O.      Rehotun D.   

      Sets_P1  Sets_P2  P1_G1  P2_G1  P1_G2  P2_G2  P1_G3  P2_G3  P1_G4  \
7168        1        3     11      5      3     11      8     11    9.0   
1           0        3     12     14      8     11      5     11    NaN   
7381        0        3      9     11      7     11      6     11    NaN   
1483        0        3      8     11      8     11      3     11    NaN   
6130        3        1     12     10     13     15     11      9   11.0   
3116        3        0     13     11     11      8     11      5    NaN   
4013        1        3     11      6      9     11      6     11    6.0   
601         2        3      7     11     11      7      5     11   11.0   
4460        1        3     10     12     11      3      9     11    9.0   
758         1        3      6     11      9     11     13     11    7.0   
3592        3        1     11      5     11      7      8     11   13.0   
5879        1        3      4     11     11      7      4     11    4.0   
7204        3        2     13     11      7     11      9     11   11.0   
37          3        0     11      9     11      8     11      7    NaN   
4753        3        0     11      7     11      6     11      6    NaN   
6841        2        3     11      5      7     11     11      3   12.0   
5358        3        1      9     11     11      7     11      8   11.0   
3827        2        3      5     11     11      7      5     11   11.0   
6083        3        2      9     11     11      7     16     14    7.0   
2504        0        3     12     14      7     11      5     11    NaN   

      P2_G4  P1_G5  P2_G5  HomeWinner  
7168   11.0    NaN    NaN           0  
1       NaN    NaN    NaN           0  
7381    NaN    NaN    NaN           0  
1483    NaN    NaN    NaN           0  
6130    9.0    NaN    NaN           1  
3116    NaN    NaN    NaN           1  
4013   11.0    NaN    NaN           0  
601     6.0    9.0   11.0           0  
4460   11.0    NaN    NaN           0  
758    11.0    NaN    NaN           0  
3592   11.0    NaN    NaN           1  
5879   11.0    NaN    NaN           0  
7204    6.0   14.0   12.0           1  
37      NaN    NaN    NaN           1  
4753    NaN    NaN    NaN           1  
6841   14.0    9.0   11.0           0  
5358    6.0    NaN    NaN           1  
3827    7.0    4.0   11.0           0  
6083   11.0   12.0   10.0           1  
2504    NaN    NaN    NaN           0

#### VALORES FALTANTES DEL DATASET

In [46]:
df.dropna()

Unnamed: 0                 Date         Player1        Player2  Sets_P1  \
5              6  2022-06-10 02:30:00       Sachuk S.   Diachenko M.        2   
7              8  2022-06-10 02:55:00  Hryshchenko O.     Sokolov S.        3   
9             10  2022-06-10 03:25:00        Kyilo V.      Sachuk S.        2   
11            12  2022-06-10 03:55:00      Sokolov S.  Chernyshov D.        3   
12            13  2022-06-10 04:05:00   Balakiriev M.     Kostiuk B.        3   
...          ...                  ...             ...            ...      ...   
7842        7843  2022-07-08 13:40:00    Yurchenko K.    Zholubak R.        2   
7846        7847  2022-07-08 13:50:00        Rehak M.      Bazany D.        3   
7847        7848  2022-07-08 13:55:00     Bielokon Y.      Yushko R.        3   
7848        7849  2022-07-08 13:55:00      Kuzmych V.         Rak S.        2   
7850        7851  2022-07-08 14:00:00    Kurylenko O.    Kuzmenko D.        2   

      Sets_P2  P1_G1  P2_G1  P1_G2  P2_G2  P1_G3  P2_G3  P1_G4  P2_G4  P1_G5  \
5           3     11      9     11      9      5     11   10.0   12.0    0.0   
7           2     11      8      7     11     11      3    9.0   11.0   11.0   
9           3      9     11      5     11     11      7   11.0    6.0    9.0   
11          2     11      7     12     10      5     11    6.0   11.0   11.0   
12          2     11      5     13     11      7     11    8.0   11.0   11.0   
...       ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
7842        3     11      9      7     11     11      6    7.0   11.0   12.0   
7846        2      4     11     11      9      4     11   11.0    5.0   11.0   
7847        2     11      6     11      8      8     11   10.0   12.0   11.0   
7848        3     11      9      3     11     13     11    8.0   11.0    5.0   
7850        3     11      7      5     11     12     10    6.0   11.0    9.0   

      P2_G5  HomeWinner  
5      11.0           0  
7       9.0           1  
9      11.0           0  
11      9.0           1  
12      8.0           1  
...     ...         ...  
7842   14.0           0  
7846    7.0           1  
7847    3.0           1  
7848   11.0           0  
7850   11.0           0  

[2518 rows x 17 columns]

In [47]:
df.fillna(0)


Unnamed: 0                 Date         Player1         Player2  \
0              1  2022-06-10 01:05:00      Marchuk M.  Polysynskyi M.   
1              2  2022-06-10 01:25:00      Sokolov S.       Sachuk S.   
2              3  2022-06-10 01:35:00      Samofal Y.     Bambutsa R.   
3              4  2022-06-10 01:55:00        Kyilo V.   Chernyshov D.   
4              5  2022-06-10 02:05:00  Polysynskyi M.      Kostiuk B.   
...          ...                  ...             ...             ...   
7846        7847  2022-07-08 13:50:00        Rehak M.       Bazany D.   
7847        7848  2022-07-08 13:55:00     Bielokon Y.       Yushko R.   
7848        7849  2022-07-08 13:55:00      Kuzmych V.          Rak S.   
7849        7850  2022-07-08 14:00:00   Bondarenko I.    Havryliuk B.   
7850        7851  2022-07-08 14:00:00    Kurylenko O.     Kuzmenko D.   

      Sets_P1  Sets_P2  P1_G1  P2_G1  P1_G2  P2_G2  P1_G3  P2_G3  P1_G4  \
0           3        1     11      8     11      6      6     11   11.0   
1           0        3     12     14      8     11      5     11    0.0   
2           3        1      4     11     11      5     11      6   11.0   
3           3        1      6     11     11      1     15     13   11.0   
4           3        0     12     10     11      9     11      6    0.0   
...       ...      ...    ...    ...    ...    ...    ...    ...    ...   
7846        3        2      4     11     11      9      4     11   11.0   
7847        3        2     11      6     11      8      8     11   10.0   
7848        2        3     11      9      3     11     13     11    8.0   
7849        0        3     10     12      8     11      4     11    0.0   
7850        2        3     11      7      5     11     12     10    6.0   

      P2_G4  P1_G5  P2_G5  HomeWinner  
0       8.0    0.0    0.0           1  
1       0.0    0.0    0.0           0  
2       9.0    0.0    0.0           1  
3       7.0    0.0    0.0           1  
4       0.0    0.0    0.0           1  
...     ...    ...    ...         ...  
7846    5.0   11.0    7.0           1  
7847   12.0   11.0    3.0           1  
7848   11.0    5.0   11.0           0  
7849    0.0    0.0    0.0           0  
7850   11.0    9.0   11.0           0  

[7851 rows x 17 columns]

### SOBRE EL DATASET

CONTEXTO ANALITICO

El Dataset con el que se trabajará a continuación corresponde a un conjunto de datos referidos a la recopilación de partidos de Tenis de mesa durante 1 mes. Cada fila representa 1 partido jugado al mejor de 5 sets. Cada set es jugado a 11 puntos, o cuando se cumple una diferencia de 2 puntos.
El archivo .csv utilizado fue descargado de la página kaggle.com.

CONTEXTO COMERCIAL

El objetivo del presente trabajo es intentar predecir la duración del partido, mediante el uso de diferentes variables, las cuales se consideran con mayor preponderancia a la hora de determinar cuanto se extenderán los partidos.

DESCRIPCIÓN

Todos los atributos del dataset son listados a continuación:
- #: Número de partido (ordenado cronológicamente)
- Player1: Nombre del primer jugador
- Player2: Nombre del segundo jugador
- Sets_P1: número de sets ganados por el primer jugador
- Sets_P2: número de sets ganados por el segundo jugador
- P1_G1: número de puntos ganados por el primer jugador en el primer set
- P2_G1: número de puntos ganados por el segundo jugador en el primer set
- P1_G2: número de puntos ganados por el primer jugador en el segundo set
- P2_G2: número de puntos ganados por el segundo jugador en el segundo set
- P1_G3: número de puntos ganados por el primer jugador en el tercer set
- P2_G3: número de puntos ganados por el segundo jugador en el tercer set
- P1_G4: número de puntos ganados por el primer jugador en el cuarto set (NA significa que no se jugó el cuarto set)
- P2_G4: número de puntos ganados por el segundo jugador en el cuarto set (NA significa que no se jugó el cuarto set)
- P1_G5: número de puntos ganados por el primer jugador en el quinto set (NA significa que no se jugó el quinto set)
- P2_G5: número de puntos ganados por el segundo jugador en el quinto set (NA significa que no se jugó el quinto set)
- HomeWinner: ganador (1 significa que ganó el Jugador 1, 0 significa que ganó el Jugador 2)

El dataset seleccionado cuenta con 7851 filas y 17 columnas

In [14]:
display(df.shape)

(7851, 17)

El dataset cuenta con los siguientes nombres en sus columnas:

In [15]:
df.columns

Index(['Unnamed: 0', 'Date', 'Player1', 'Player2', 'Sets_P1', 'Sets_P2',
       'P1_G1', 'P2_G1', 'P1_G2', 'P2_G2', 'P1_G3', 'P2_G3', 'P1_G4', 'P2_G4',
       'P1_G5', 'P2_G5', 'HomeWinner'],
      dtype='object')

El dataset cuenta con los siguientes tipos de datos en sus columnas:

In [17]:
df.dtypes

Unnamed: 0      int64
Date           object
Player1        object
Player2        object
Sets_P1         int64
Sets_P2         int64
P1_G1           int64
P2_G1           int64
P1_G2           int64
P2_G2           int64
P1_G3           int64
P2_G3           int64
P1_G4         float64
P2_G4         float64
P1_G5         float64
P2_G5         float64
HomeWinner      int64
dtype: object

La información proporcionada es la siguiente:

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7851 entries, 0 to 7850
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  7851 non-null   int64  
 1   Date        7851 non-null   object 
 2   Player1     7851 non-null   object 
 3   Player2     7851 non-null   object 
 4   Sets_P1     7851 non-null   int64  
 5   Sets_P2     7851 non-null   int64  
 6   P1_G1       7851 non-null   int64  
 7   P2_G1       7851 non-null   int64  
 8   P1_G2       7851 non-null   int64  
 9   P2_G2       7851 non-null   int64  
 10  P1_G3       7851 non-null   int64  
 11  P2_G3       7851 non-null   int64  
 12  P1_G4       5448 non-null   float64
 13  P2_G4       5448 non-null   float64
 14  P1_G5       2518 non-null   float64
 15  P2_G5       2518 non-null   float64
 16  HomeWinner  7851 non-null   int64  
dtypes: float64(4), int64(10), object(3)
memory usage: 1.0+ MB


La variable que se busca predecir es la llamada "Ganador del Partido" a=HomeWinner

Las variables consideradas relevantes para predecir la duración del partido son:
 - a1=Sets_P1: corrrespondiente al número de sets ganados por el primer jugador. Es la variable más relevante a la hora de la evaluación, siendo que si este valor supera los 2 sets se considera ganador del partido.
 - a2=P1_G4: número de puntos ganados por el primer jugador en el cuarto set. Se ha estudiado que la manera en la que se juega el 4to set, y consecuentemente los puntos ganados en el mismo, son un factor clave a la hora de evaluar la manera de afrontar el set definitivo en caso de quedar el marcador igualado.
 - a3=1_G5: número de puntos ganados por el primer jugador en el quinto set. Es determinante ya que es el set decisivo, pero puede no ser siempre una opcion a evaluar, en caso de que no se llegue a esa instancia.


In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Rectangle
import matplotlib as mpl

In [7]:
pd.read_csv('/workspaces/CoderHouse_CH-32845/data/raw/waterquality.csv')

aluminium ammonia  arsenic  barium  cadmium  chloramine  chromium  \
0          1.65    9.08     0.04    2.85    0.007        0.35      0.83   
1          2.32   21.16     0.01    3.31    0.002        5.28      0.68   
2          1.01   14.02     0.04    0.58    0.008        4.24      0.53   
3          1.36   11.33     0.04    2.96    0.001        7.23      0.03   
4          0.92   24.33     0.03    0.20    0.006        2.67      0.69   
...         ...     ...      ...     ...      ...         ...       ...   
7994       0.05    7.78     0.00    1.95    0.040        0.10      0.03   
7995       0.05   24.22     0.02    0.59    0.010        0.45      0.02   
7996       0.09    6.85     0.00    0.61    0.030        0.05      0.05   
7997       0.01      10     0.01    2.00    0.000        2.00      0.00   
7998       0.04    6.85     0.01    0.70    0.030        0.05      0.01   

      copper  flouride  bacteria  ...   lead  nitrates  nitrites  mercury  \
0       0.17      0.05      0.20  ...  0.054     16.08      1.13    0.007   
1       0.66      0.90      0.65  ...  0.100      2.01      1.93    0.003   
2       0.02      0.99      0.05  ...  0.078     14.16      1.11    0.006   
3       1.66      1.08      0.71  ...  0.016      1.41      1.29    0.004   
4       0.57      0.61      0.13  ...  0.117      6.74      1.11    0.003   
...      ...       ...       ...  ...    ...       ...       ...      ...   
7994    0.03      1.37      0.00  ...  0.197     14.29      1.00    0.005   
7995    0.02      1.48      0.00  ...  0.031     10.27      1.00    0.001   
7996    0.02      0.91      0.00  ...  0.182     15.92      1.00    0.000   
7997    0.09      0.00      0.00  ...  0.000      0.00      0.00    0.000   
7998    0.03      1.00      0.00  ...  0.182     15.92      1.00    0.000   

      perchlorate  radium  selenium  silver  uranium  is_safe  
0           37.75    6.78      0.08    0.34     0.02        1  
1           32.26    3.21      0.08    0.27     0.05        1  
2           50.28    7.07      0.07    0.44     0.01        0  
3            9.12    1.72      0.02    0.45     0.05        1  
4           16.90    2.41      0.02    0.06     0.02        1  
...           ...     ...       ...     ...      ...      ...  
7994         3.57    2.13      0.09    0.06     0.03        1  
7995         1.48    1.11      0.09    0.10     0.08        1  
7996         1.35    4.84      0.00    0.04     0.05        1  
7997         0.00    0.00      0.00    0.00     0.00        1  
7998         1.35    4.84      0.00    0.04     0.05        1  

[7999 rows x 21 columns]

In [6]:
mpl.style.use('bmh')

NameError: name 'mpl' is not defined

## hola

# hola

**hola**

*hola*

- fgdfgdfg
- sdfsdf

* zxczxc
* zxczxc

1. asdasds
    asdasd

[link a youtube](https://youtube.com)

---

$2+2 = 4$

voy a colocar una formula $$2+2 =4$$